sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/display/DisplayData.java

```diff
@@ -182,309 +183,79 @@ private static String namespaceOf(ClassForDisplay clazz) {
+     * item display item belongs to.
      */
-    ItemBuilder withLinkUrl(@Nullable String url);
-
-    /**
-     * Adds an explicit namespace to the most-recently added display data. The namespace
-     * and key uniquely identify the display data.
-     *
-     * <p>Leaving the namespace unspecified will default to the registering instance's class.
-     */
-    ItemBuilder withNamespace(Class<?> namespace);
+    @Nullable
+    @JsonGetter("namespace")
+    public abstract String getNamespace();
 
     /**
-     * Adds an explicit namespace to the most-recently added display data. The namespace
-     * and key uniquely identify the display data.
-     *
-     * <p>Leaving the namespace unspecified will default to the registering instance's class.
+     * The key for the display item. Each display item is create with a key and value
+     * via {@link DisplayData#item).
      */
-    ItemBuilder withNamespace(ClassForDisplay namespace);
-  }
-
-  /**
-   * A display data item. DisplayData items are registered via {@link Builder#add} within
-   * {@link HasDisplayData#populateDisplayData} implementations. Each metadata item is uniquely
-   * identified by the specified key and namespace generated from the registering component's
-   * class name.
-   */
-  public static class Item {
-    private final String key;
-    private final String ns;
-    private final Type type;
-    private final Object value;
-    private final Object shortValue;
-    private final String label;
-    private final String url;
-
-    private static Item create(String nsClass, String key, Type type, Object value) {
-      FormattedItemValue formatted = type.format(value);
-      return new Item(
-          nsClass, key, type, formatted.getLongValue(), formatted.getShortValue(), null, null);
-    }
-
-    private Item(
-        String namespace,
-        String key,
-        Type type,
-        Object value,
-        Object shortValue,
-        String url,
-        String label) {
-      this.ns = namespace;
-      this.key = key;
-      this.type = type;
-      this.value = value;
-      this.shortValue = shortValue;
-      this.url = url;
-      this.label = label;
-    }
-
-    @JsonGetter("namespace")
-    public String getNamespace() {
-      return ns;
-    }
-
     @JsonGetter("key")
-    public String getKey() {
-      return key;
-    }
+    public abstract String getKey();
 
     /**
      * Retrieve the {@link DisplayData.Type} of display data. All metadata conforms to a
      * predefined set of allowed types.
      */
     @JsonGetter("type")
-    public Type getType() {
-      return type;
-    }
+    public abstract Type getType();
 
     /**
-     * Retrieve the value of the metadata item.
+     * Retrieve the value of the display item. The value is translated from input to
+     * {@link DisplayData#item} into a format suitable for display, using
+     * {@link Type#format(Object)}.
```

Comment:
I've removed the reference to `Type#format`, as this method and `safeFormat` are both internal and implementation details.

`Type#format` and `Type#safeFormat` return a `FormattedItemValue`, which is a container holding both the translated `value` and `shortValue`.